lib/ecto/changeset.ex

```diff
@@ -729,6 +738,9 @@ defmodule Ecto.Changeset do
         case Map.get(types, key) do
           {:embed, embed} ->
             {key, Ecto.Embedded.apply_changes(embed, value)}
+          {:assoc, _assoc} ->
+            raise ArgumentError, "associations should be removed from changeset" <>
```

Comment:
We will be calling `Repo.insert!/update!/delete!` for the assocs, and using the returned model from there, and not applying the changes from the changeset (repo functions do that already for us)